### DISCUSSION TITLE
Just touching base

### DISCUSSION THREAD

**Discussion by fabnavigator** [zwave-js/node-red-contrib-zwave-js#277] [Q&A] [answered]
Hi @marcus-j-davies,

It's been a while, and I was just wondering how you have been? All of my z-wave devices are working well, so no complaints here. I was wondering if I should learn something about firmware updating. I see that there is something in the controller node about that.

Best reagrds,

Bill

**Comment by fabnavigator** [comment]
Oh, and I recently removed security from almost all of my devices. Less battery drain, and so much easier to sniff.

**Reply by marcus-j-davies** [reply]
Hi @fabnavigator 

I'm good - still alive 😅

I have ZWave devices that are security centric - Security Entry Pin Pads that wont respond unless using S2 for example.

As for the firmware thing - ZWave JS offers a service that locates the most recent Firmware for your device, and offers to update it
(I of course implemented it).

This is free for personal use, but commercial use require a license i.e - if my module is in use in a  commercial setting - a license is required (as per Zwave JS Firmware Update Service) 

https://github.com/zwave-js/firmware-updates

under normal circumstance, if there is no reason to update the firmware on your devices - its best left alone. 

**What am I doing?**
Currently involved in 2 large projects (Freelance + Employment Duties) has taken up best part of 2023 thus far.

**Where Have I been.**
I was on your side of the pond not long  ago, Texas + NYC  - to visit this guy called @crxporter.

**What I need to do.**
LOTS!!
I still have V9 in sights  - I have not forgotten it, just time is limited presently! 😅

**Reply by marcus-j-davies** [reply]
In fact - I may soon push an update, to bring the module up to the latest Major Driver version - but it wont benefit with any enhancements of said version, until V9

**EDIT:**
Sticking this here for my self:
https://zwave-js.github.io/node-zwave-js/#/getting-started/migrating-to-v11

**Reply by fabnavigator** [reply]
@marcus-j-davies, I'm glad you are still alive! It looks like I need a license key. Is that somethingt that you can provide? I have a device that has some configuration parameters that aren't available in the current firmware.

**Reply by marcus-j-davies** [reply, marked as answer]
You only need a license key, if using the Update Service - if you have access to the Firmware file itself - you can use the Non Service Provided option - in the Side Panel, there is a FW updater option.

if you want to use the service - use **NON-COMMERCIAL** as a license key, for personal use.
if you use it commercially  - email info@zwave-js.io to obtain one.

Note: The service is still young, so may not locate any FW files automatically  - they are provided by Vendors, as and when they are available (and of course if they provide them to us)

![Screenshot 2023-07-24 at 19 43 16](https://github.com/zwave-js/node-red-contrib-zwave-js/assets/55892693/6fccba13-a036-4bf7-9e29-73daea9986d3)
